contracts/apps/AragonApp.sol

```diff
@@ -3,15 +3,29 @@ pragma solidity ^0.4.18;
 import "./AppStorage.sol";
 import "../common/Initializable.sol";
 import "../evmscript/EVMScriptRunner.sol";
+import "../acl/ACLSyntaxSugar.sol";
 
 
-contract AragonApp is AppStorage, Initializable, EVMScriptRunner {
+contract AragonApp is AppStorage, Initializable, ACLSyntaxSugar, EVMScriptRunner {
     modifier auth(bytes32 _role) {
-        require(canPerform(msg.sender, _role));
+        require(canPerform(msg.sender, _role, new uint256[](0)));
         _;
     }
 
-    function canPerform(address _sender, bytes32 _role) public view returns (bool) {
-        return address(kernel) == 0 || kernel.hasPermission(_sender, address(this), _role, new bytes(0));
+    modifier authP(bytes32 _role, uint256[] params) {
+        require(canPerform(msg.sender, _role, params));
+        _;
+    }
+
+    function canPerform(address _sender, bytes32 _role, uint256[] params) public view returns (bool) {
+        bytes memory how; // no need to init memory as it is never used
```

Comment:
We now have two slightly different ways of doing this `uint[]` to `bytes` conversion (see `canPerform()` in `Kernel`), so I'm wondering if it'd be nice to add this as `toBytes()` somewhere. Maybe we can move some of the more general helpers in `ScriptHelpers` (e.g. `uint256At`, `addressAt`, `uint32At`, `locationOf`) to another library like `BytesHelpers`?

Reply:
Good idea, making this its own issue